js/crm.datepicker.js

```diff
@@ -33,21 +34,27 @@
       }
       if (settings.time !== false) {
         $timeField = $('<input>').insertAfter($dataField);
+        placeholder = settings.timePlaceholder || $dataField.attr('time-placeholder');
         CRM.utils.copyAttributes($dataField, $timeField, ['class', 'disabled']);
         $timeField
           .addClass('crm-form-text crm-form-time')
-          .attr('placeholder', $dataField.attr('time-placeholder') === undefined ? '\uf017' : $dataField.attr('time-placeholder'))
-          .attr('aria-label', $dataField.attr('time-placeholder') === undefined ? ts('Time') : $dataField.attr('time-placeholder'))
+          // Set default placeholder as clock icon (`fa-clock` is Unicode f017)
+          .attr('placeholder', placeholder === undefined ? '\uf017' : placeholder)
+          .attr('aria-label', placeholder === undefined ? ts('Time') : placeholder)
           .change(updateDataField)
           .timeEntry({
             spinnerImage: '',
             show24Hours: settings.time === true || settings.time === undefined ? CRM.config.timeIs24Hr : settings.time == '24'
           });
+        if (!placeholder) {
+          $timeField.addClass('crm-placeholder-right');
```

Comment:
You could instead switch this (and the spot on line 82) to `crm-placeholder-icon` to match the CSS, but I like having `-right` in the class to be clear that we're aligning it to the right.  I suppose if you want to be real fancy you could set two classes: `crm-placeholder-icon` that sets it to be Font Awesome and `crm-placeholder-right` that aligns the placeholder to the right.